### DISCUSSION TITLE
Controller + React-Select + useState woes

### DISCUSSION THREAD

**Discussion by paambaati** [react-hook-form/react-hook-form#3841] [Q&A]
I'm trying to build a `Controller`-ified wrapper component for `react-select`, and I'm not able to use `value` set to a state variable to correctly reflect the component's form value.

See https://codesandbox.io/s/react-hook-form-controller-react-select-state-variable-troubles-sl3vv

**What I'm trying to do**

Have a `value` prop on my component driven by state. So when `value` changes, the component's `setValue()` should be called correctly with the new value.

**The problem**
In the example linked above, you'll notice that if you click on the 'Set Blue' button, the state variable is correctly set, but the form component's value isn't set.

What am I doing wrong?

**Comment by bluebill1049** [comment]
why not just use `setValue` API?

**Reply by bluebill1049** [reply]
also you can always use `useController`, so everything is at your own hand.

**Reply by paambaati** [reply]
If I had to depend on `setValue`, I’d have no way of tracking changes to the state variable. In the linked example, if the `color` value changes, `setValue` needs to be called.

Is this not how `Controller` works?

**Reply by bluebill1049** [reply]
prop.onChange is bascially setValue

**Reply by paambaati** [reply]
That’s what I’d expected too, but the CSB doesn’t seem to work.

**Reply by bluebill1049** [reply]
you are using useState, useState doesn't have context with hook form.

**Reply by paambaati** [reply]
Right. Can you show me how to do it with react-select though? A lot of the examples either use the outdated `as` prop or don’t show an integration with react-select.

**Reply by bluebill1049** [reply]
sure, i can, but why do you need to use `useState`?

**Reply by paambaati** [reply]
Because I want to change the value on a user interaction. So far, I’ve only had to support defaultValue, and that’s been working fine for me so far.

Now, I have a new user interaction (something like “auto fill this form with these values”), so I need to change or set values on the form. And those variables happen to be `useState` variables.

**Reply by bluebill1049** [reply]
why not do that from the place where you do the `useState`? or otherwise you can trigger at useEffect
